src/entities/smaOracle.ts

```diff
@@ -77,27 +83,52 @@ export default class SMAOracle implements OracleClass<SMAOracleContract> {
 			oracleInfo.provider
 		);
 		this._contract = contract;
-		const [updateInterval, numPeriods] = await Promise.all([
+		const [updateInterval, numPeriods, underlyingOracleAddress] = await Promise.all([
 			contract.updateInterval(),
-			contract.numPeriods()
+			contract.numPeriods(),
+			contract.oracle()
 		])
+		this._underlyingOracle = ChainlinkOracleWrapper__factory.connect(
+			underlyingOracleAddress,
+			this.provider
+		);
 		this.updateInterval = updateInterval.toNumber();
 		this.numPeriods = numPeriods.toNumber();
 	}
 
-
+	/**
+	 * get the latest SMA price
+	 * @returns the average of the most recent price data points
+	 */
 	public getPrice: () => Promise<BigNumber> = async () => {
 		if (!this._contract) {
 			throw Error("Failed to fetch oracle price: Contract not defined");
 		}
 		try {
 			const price = await this._contract.getPrice();
+			// OracleWrapper always scales feed decimals to 18 places
 			return new BigNumber(ethers.utils.formatEther(price))
 		} catch (err) {
 			throw Error(`Failed to fetch oracle price: ${err}`)
 		}
 	}
 
+		/**
+	 * get the current underlying spot price
+	 * @returns the current price reported by the underlying price oracle
+	 */
+	public getSpotPrice: () => Promise<BigNumber> = async () => {
+		if (!this._underlyingOracle) {
+			throw Error("Failed to fetch sma spot price: underlying oracle not defined");
+		}
+		try {
+			const price = await this._underlyingOracle.getPrice();
```

Comment:
is it always wei?

Reply:
yep https://github.com/tracer-protocol/perpetual-pools-contracts/blob/pools-v2/contracts/implementation/ChainlinkOracleWrapper.sol#L63